.github/container/create-distribution.sh

```diff
@@ -78,18 +95,36 @@ if [[ $# -ge 1 ]]; then
     usage 1
 fi
 
-set -euox pipefail
-SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+set -eou pipefail
+# readlink -f $(pwd) is cross-platform way to ensure /tmp gets resolved correctly on macos
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && readlink -f $(pwd) )
 
-INSTALLED_DIR=${INSTALLED_DIR:-/opt/t5x}
-DISTRIBUTION_BASE_REF=${DISTRIBUTION_BASE_REF:-HEAD}
-MIRROR_GIT_URL=${MIRROR_GIT_URL:-https://github.com/nvjax-svc-0/t5x.git}
-
-if [[ -z "${INSTALLED_DIR}" ]]; then
-  echo "[ERROR]: Need to specify -d/--dir"
+if [[ -z "$MANIFEST" || -z "$PACKAGE" ]]; then
+  echo "--manifest and --package must be provided"
   usage 1
 fi
 
+BASE_DIR=${BASE_DIR:-/opt}
+CLEAN_PATCHES=${CLEAN_PATCHES:-0}
+UPSTREAM_URL=$(yq e ".${PACKAGE}.url" $MANIFEST)
+# The tracking_ref is interpreted as the default "main" branch and all patches are 
+# assumed to be rooted on a sha on the tracking_ref's history
+TRACKING_REF=$(yq e ".${PACKAGE}.tracking_ref" $MANIFEST)
+INSTALLED_DIR=${OVERRIDE_INSTALL_DIR:-/opt/${PACKAGE}}
+MIRROR_GIT_URL=$(yq e ".${PACKAGE}.mirror_url // \"\"" $MANIFEST)
+EXTRA_DIR=$(yq e ".${PACKAGE}.extra_dir // \"\"" $MANIFEST)
+
+SKIP_APPLY=${SKIP_APPLY:-0}
+GEN_PATCH_DIR=${GEN_PATCH_DIR:-$SCRIPT_DIR/patches/$PACKAGE}
```

Comment:
I originally left that in to give me a way to test this script without adding a cmdline arg since this script edits things in place. I no longer need that since I just copy the script to test it (see `rosetta/tests/*`), so I can remove the option to override it